deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -191,23 +193,25 @@ static String durationToHeaderValue(long durationNanos) {
     @Nullable
     @VisibleForTesting
     static Long tryParseSecondsToNanoseconds(String value) {
-        if (Strings.isNullOrEmpty(value)) {
-            return null;
+        NumberFormatException exception = null;
+        String normalized = Strings.nullToEmpty(value).trim();
+        if (!normalized.isEmpty() && decimalMatcher.matchesAllOf(normalized)) {
+            try {
+                double seconds = Double.parseDouble(normalized);
+                return (long) (seconds * 1e9d);
+            } catch (NumberFormatException e) {
+                exception = e;
+            }
         }
 
-        Double seconds = Doubles.tryParse(value);
-        if (seconds == null) {
-            if (log.isWarnEnabled()) {
-                if (logLimiter.tryAcquire()) {
-                    log.warn("Failed to parse 'Expect-Within' header value", SafeArg.of("value", value));
-                } else if (log.isDebugEnabled()) {
-                    log.debug("Failed to parse 'Expect-Within' header value", SafeArg.of("value", value));
-                }
+        if (log.isWarnEnabled()) {
+            if (logLimiter.tryAcquire()) {
+                log.warn("Failed to parse 'Expect-Within' header value", UnsafeArg.of("value", value));
+            } else if (log.isDebugEnabled()) {
+                log.debug("Failed to parse 'Expect-Within' header value", UnsafeArg.of("value", value), exception);
```

Comment:
Given we've defined this header and it's not used elsewhere to my knowledge, it should be safe to log, even when values are malformed. Otherwise we're not going to have a great time debugging.